apps/form-system/web/src/utils/validation.ts

```diff
@@ -0,0 +1,112 @@
+import { ApplicationState, FieldTypesEnum } from '@island.is/form-system/ui'
+import { FormSystemField, FormSystemValue } from '@island.is/api/schema'
+
+export const validateScreen = (state: ApplicationState): string[] => {
```

Comment:
I think you should look into using react-hook-form for the form state and validation and error handling, it looks like we are recreating a lot of the functionality it provides out of the box.